src/pnm/encoder.rs

```diff
@@ -636,9 +637,9 @@ mod tests {
         let data: [u8; 12] = [0, 0, 0, 1, 1, 1, 255, 255, 255, 0, 0, 0];
 
         let header = ArbitraryHeader {
-            width: 2,
-            height: 2,
-            depth: 3,
+            width: 3,
+            height: 4,
+            depth: 1,
```

Comment:
Does that mean `depth: 3` no longer passes? I would have expected `ColorType::Unknown(24)` or something but still working. Or with an extra count for the number of color planes, note that `depth` in the context of `pam` refers actually to the number of color planes.. Weird but that could keep it working.

Reply:
This is a consequence of the `num_channels(ColorType::Unknown(..)) = 1` change from above. I'm really not sure why we support non-standard encodings like this, but if we want to get the old behavior back the best way would just be to tell the encoder that the input data has 3 8-bit channels by passing `ColorType::RGB`. The ArbitraryTuplType::Custom argument already causes the output header from including the true input color type.

Reply:
That's because `pnm` completely separate 'color tuple type' and 'channel count/depth' information. The `ArbitraryTuplType` only sets the former and the encoder only ensures that that information is consistent with the `ColorType` for non-`Custom` variants.

Reply:
But that also means that including the true input color type is only possible if setting arbitrary color channel count and bit-depth information is also available. I had previously used `Palette` to at least allow a count of `3`. It's not strictly necessary.

Reply:
The current code checks that `depth == num_components(color)` even for Custom variants, which just happened to work because Palette previously was declared to have 3 color channels. That might be fixable but I'd like to hold off making more intrusive changes to the pnm decoder for now